apis/database/v1alpha1/redis/redis_types.go

```diff
@@ -0,0 +1,126 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+
+// RedisInstance is the Schema for the redisinstances API
+// An RedisInstance is a managed resource that represents an Redis instance.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.atProvider.dbInstanceStatus"
+// +kubebuilder:printcolumn:name="ENGINE",type="string",JSONPath=".spec.forProvider.engine"
+// +kubebuilder:printcolumn:name="VERSION",type="string",JSONPath=".spec.forProvider.engineVersion"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,alibaba}
+type RedisInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RedisInstanceSpec   `json:"spec,omitempty"`
+	Status RedisInstanceStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// RedisInstanceList contains a list of RedisInstance
+type RedisInstanceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []RedisInstance `json:"items"`
+}
+
+// RedisInstanceSpec defines the desired state of RedisInstance
+type RedisInstanceSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+	ForProvider       RedisInstanceParameters `json:"forProvider"`
+}
+
+// Redis instance states.
+const (
+	// The instance is healthy and available
+	RedisInstanceStateRunning = "Normal"
+	// The instance is being created. The instance is inaccessible while it is being created.
+	RedisInstanceStateCreating = "Creating"
+	// The instance is being deleted.
+	RedisInstanceStateDeleting = "Flushing"
+)
+
+// RedisInstanceStatus defines the observed state of RedisInstance
+type RedisInstanceStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+	AtProvider          RedisInstanceObservation `json:"atProvider,omitempty"`
+}
+
+// RedisInstanceParameters define the desired state of an Redis instance.
+type RedisInstanceParameters struct {
```

Comment:
It's might be better to change parameters names same to those listed in Redis instance creation api specification https://www.alibabacloud.com/help/doc-detail/60873.htm?spm=a2c63.p38356.879954.6.2a544542iP8ySE#t3181.html.

Also refer to https://github.com/crossplane/crossplane/blob/146bbc14/design/one-pager-managed-resource-api-design.md#high-fidelity